pkg/users/authorized-keys.js

```diff
@@ -97,34 +95,23 @@
             return update_keys([], tag);
 
         parse_pubkeys(content)
-                .done(function(keys) {
-                    update_keys(keys, tag);
-                });
+                .then(keys => update_keys(keys, tag));
     }
 
     self.add_key = function(key) {
-        const df = cockpit.defer();
-        df.notify(_("Validating key"));
-        parse_pubkeys(key)
-                .done(function(keys) {
+        return parse_pubkeys(key)
+                .then(keys => {
                     const obj = keys[0];
                     if (obj && obj.valid) {
-                        df.notify(_("Adding key"));
-                        cockpit
+                        return cockpit
                                 .script(adder, [user_name, home_dir], { superuser: "try", err: "message" })
                                 .input(obj.raw + "\n")
-                                .done(function() {
-                                    df.resolve();
-                                })
-                                .fail(function(ex) {
-                                    df.reject(_("Error saving authorized keys: ") + ex);
-                                });
+                                // eslint-disable-next-line prefer-promise-reject-errors
+                                .catch(ex => Promise.reject(_("Error saving authorized keys: ") + ex));
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18488-20230310-163728-a6731aad-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)